src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs

```diff
@@ -330,16 +330,16 @@ private bool DictionaryEntriesEqual(DictionaryEntry x, DictionaryEntry y, ref To
 
         private bool StringsEqual(string x, string y)
         {
-            string s1 = caseInsensitive ? x.ToLower() : x;
-            string s2 = caseInsensitive ? y.ToLower() : y;
+            string s1 = caseInsensitive ? x.ToUpperInvariant() : x;
+            string s2 = caseInsensitive ? y.ToUpperInvariant() : y;
 
             return s1.Equals(s2);
```

Comment:
@Lette Looks like we want to keep the original behavior, so let's not make any changes at all. `string.Equals(,,)` would be an improvement but orthogonal to the purposes of this PR.